reqh/reqh_service.c

```diff
@@ -459,7 +459,8 @@ M0_INTERNAL void m0_reqh_service_fini(struct m0_reqh_service *service)
 	M0_PRE(service != NULL && m0_reqh_service_bob_check(service));
 
 	M0_ASSERT(m0_fom_domain_is_idle_for(service));
-	m0_locality_lockers_free(service->rs_fom_key);
+	if (m0_reqh_rpc_mach_tlist_is_empty(&service->rs_reqh->rh_rpc_machines))
+		m0_locality_lockers_free(service->rs_fom_key);
```

Comment:
I have a doubt:
If we have this check, the locker will not be freed when we still have valid RPC machines.
But when all RPC machines are removed, there is no callback to free the key.
This will introduced key leakage.

Reply:
this solution was workaround, as discussed we should not post terminate request when service is STOPPED, so ignore this solution